DirectoryLoader slicing

amovfx opened the pull request:
DirectoryLoader can now return a random sample of files in a directory.
Parameters added are:
sample_size
randomize_sample
sample_seed


@rlancemartin, @eyurtsev

Commit:
Added ability to randomly sample files in a dir.

amovfx commented:
Found this helpful in prototyping in notebooks on large data sets.

eyurtsev commented on libs/langchain/langchain/document_loaders/directory.py:
```diff
@@ -55,6 +60,8 @@ def __init__(
             use_multithreading: Whether to use multithreading. Defaults to False.
             max_concurrency: The maximum number of threads to use. Defaults to 4.
+            sample_size: The maximum number of files you would like to load from the directory.
+            randomize_sample: Suffle the files to get a random sample.
```
Could you add documentation for sample_seed

amovfx replied:
Sorry for missing that sure thing.

eyurtsev commented on libs/langchain/langchain/document_loaders/directory.py:
```diff
@@ -39,6 +41,9 @@ def __init__(
         show_progress: bool = False,
         use_multithreading: bool = False,
         max_concurrency: int = 4,
+        sample_size: Union[int, None] = None,
```
Could you add `*` here

```suggestion
        *,
        sample_size: Union[int, None] = None,
```

amovfx replied:
Will do. thanks for the review.

Commit:
PR review

added documentation for sample_seed
forced keyword arguments on sample_seed, sample_size, and randomize_sample.
sample_size defaults to 0

baskaryan committed:
lint

baskaryan commented:
thanks @amovfx !
